_playwright-tests/tests/API/SearchEnvFeature.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { test, expect, RepositoriesApi, EnvironmentsApi, ApiRepositoryResponse } from 'test-utils';
+import { poll, randomName } from 'test-utils/helpers';
+import { ListRepositoriesRequest } from 'test-utils/client';
+
+/**
+ * Test the API support for list / search environments using hardcoded EPEL repository
+ * This test validates that environment search endpoints work correctly for both URL and UUID searches.
+ */
+
+test.describe('Search Environment Endpoint', () => {
+  const testRepoUrl = 'https://dl.fedoraproject.org/pub/epel/10/Everything/x86_64/';
+  const keyword = 'kde';
+
+  test('Environment search with EPEL repository', async ({ client }) => {
+    const repositoriesApi = new RepositoriesApi(client);
+    const environmentsApi = new EnvironmentsApi(client);
+    const repoName = `SearchEnvFeature-${randomName()}`;
+
+    let testRepo: ApiRepositoryResponse;
+
+    await test.step('Check if EPEL repository already exists or create it', async () => {
+      const existingRepos = await repositoriesApi.listRepositories({
+        url: testRepoUrl,
+        origin: 'external',
+      } as ListRepositoriesRequest);
+
+      if (existingRepos.data && existingRepos.data.length > 0) {
+        testRepo = existingRepos.data[0];
+        if (testRepo.status === 'Pending') {
+          const waitWhilePending = (resp: ApiRepositoryResponse) => resp.status === 'Pending';
+          const getRepository = () =>
+            repositoriesApi.getRepository({
+              uuid: testRepo.uuid!,
+            });
+          const updatedRepo = await poll(getRepository, waitWhilePending, 30);
+          testRepo = updatedRepo;
+        }
+      } else {
+        const addedRepo = await repositoriesApi.createRepository({
```

Comment:
we don't want to create this repo, if it doesn't exist i think we should just fail the test here. eventually we won't be able to create EPEL repos anyway